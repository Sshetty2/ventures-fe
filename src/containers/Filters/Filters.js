import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

export const Filters = ({ events }) => {
  const [eventType, updateType] = useState('');
  const [month, updateMonth] = useState('');
  const [state, updateState] = useState('');

  const setFilters = (filter) => {
    const uniqueFilters = events.reduce((acc, event) => {
      if (!acc.includes(event.attributes[filter])) {
        acc.push(event.attributes[filter])
      }
      return acc
    }, []);
    return uniqueFilters.map(item => {
      const formattedItem = filter === 'event_type' ? formatType(item) : item;
      return (<option key={item} value={item}>{formattedItem}</option>)
    })
  }

  const setMonthFilter = () => {
    const months = ['January', 'February', 'March', 'April', 'May', 'June', 'July', 'August', 'September', 'October', 'November', 'December'];
    return months.map((month, index) => {
      const matchingEvents = events.filter(event => {
        const eventMonth = event.attributes.start_date.split('-')[0];
        return parseInt(eventMonth) === index + 1
      })
      return (<option key={month} value={month}>{month} ({matchingEvents.length})</option>)
    })
  }

  const formatType = (item) => {
    let itemArr = item.split('');
    itemArr[0] = itemArr[0].toUpperCase();
    return itemArr.join('');
  }

  if (events.length) {
    return (
      <div className="Filters">
        <h3 className="filters-title">FILTER EVENTS</h3>
        <select
          name="event_type"
          value={eventType}
          onChange={(e) => updateType(e.target.value)}
        >
          <option key='Type' value=''>Event Type</option>
          {setFilters('event_type')}
        </select>
        <select
          name="month"
          value={month}
          onChange={(e) => updateMonth(e.target.value)}
        >
          <option key='Month' value=''>Month</option>
          {setMonthFilter()}
        </select>
        <select
          name="state"
          value={state}
          onChange={(e) => updateState(e.target.value)}
        >
          <option key='State'value=''>State</option>
          {setFilters('state')}
        </select>
      </div>
    );
  } else {
    return(<div></div>)
  }
}

export const mapStateToProps = (state) => ({
  events: state.events,
});

export default connect(mapStateToProps)(Filters);

Filters.propTypes = {
  events: PropTypes.array
}