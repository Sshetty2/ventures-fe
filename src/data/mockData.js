export const mockEvents = {
  data: [
    {
      id: "1",
      type: "event",
      attributes: {
        name: "Shelf Road Craggin' Classic",
        city: "Canon Springs",
        state: "CO",
        event_type: "climbing",
        price: 35,
        start_date: "10-19-2018",
        end_date: "10-21-2018",
        description:
          "This October 19-21, head on down to the best limestone sport cragging Colorado has to offer, and with the American Alpine Club and CAMP USA, and be treated to a weekend of climbing, community, clinics, contests, beer, slideshows, films, giveaways, stewardship and fun.",
        event_url: "https://americanalpineclub.org/shelf-road-cc",
        image_url:
          "https://static1.squarespace.com/static/55830fd9e4b0ec758c892f81/57bc7284f7e0abe0efbb523e/59417dc21b631bad37b19a74/1497464437416/_MG_1604.jpg",
        video_url: "https://www.youtube.com/embed/Cnjv_Brs2Bo"
      }
    },
    {
      id: "2",
      type: "event",
      attributes: {
        name: "Hueco Rock Rodeo",
        city: "Canon Springs",
        state: "CO",
        event_type: "climbing",
        price: 35,
        start_date: "10-19-2018",
        end_date: "10-21-2018",
        description:
          "This October 19-21, head on down to the best limestone sport cragging Colorado has to offer, and with the American Alpine Club and CAMP USA, and be treated to a weekend of climbing, community, clinics, contests, beer, slideshows, films, giveaways, stewardship and fun.",
        event_url: "https://americanalpineclub.org/shelf-road-cc",
        image_url:
          "https://static1.squarespace.com/static/55830fd9e4b0ec758c892f81/57bc7284f7e0abe0efbb523e/59417dc21b631bad37b19a74/1497464437416/_MG_1604.jpg",
        video_url: "https://www.youtube.com/embed/Cnjv_Brs2Bo"
      }
    },
    {
      id: "3",
      type: "event",
      attributes: {
        name: "Flash Foxy Women's Climbing Festival",
        city: "Canon Springs",
        state: "CO",
        event_type: "climbing",
        price: 35,
        start_date: "10-19-2018",
        end_date: "10-21-2018",
        description:
          "This October 19-21, head on down to the best limestone sport cragging Colorado has to offer, and with the American Alpine Club and CAMP USA, and be treated to a weekend of climbing, community, clinics, contests, beer, slideshows, films, giveaways, stewardship and fun.",
        event_url: "https://americanalpineclub.org/shelf-road-cc",
        image_url:
          "https://static1.squarespace.com/static/55830fd9e4b0ec758c892f81/57bc7284f7e0abe0efbb523e/59417dc21b631bad37b19a74/1497464437416/_MG_1604.jpg",
        video_url: "https://www.youtube.com/embed/Cnjv_Brs2Bo"
      }
    },
    {
      id: "4",
      type: "event",
      attributes: {
        name: "Flash Foxy Summerfest",
        city: "Canon Springs",
        state: "CO",
        event_type: "climbing",
        price: 35,
        start_date: "10-19-2018",
        end_date: "10-21-2018",
        description:
          "This October 19-21, head on down to the best limestone sport cragging Colorado has to offer, and with the American Alpine Club and CAMP USA, and be treated to a weekend of climbing, community, clinics, contests, beer, slideshows, films, giveaways, stewardship and fun.",
        event_url: "https://americanalpineclub.org/shelf-road-cc",
        image_url:
          "https://static1.squarespace.com/static/55830fd9e4b0ec758c892f81/57bc7284f7e0abe0efbb523e/59417dc21b631bad37b19a74/1497464437416/_MG_1604.jpg",
        video_url: "https://www.youtube.com/embed/Cnjv_Brs2Bo"
      }
    },
    {
      id: "5",
      type: "event",
      attributes: {
        name: "International Climber's Festival",
        city: "Canon Springs",
        state: "CO",
        event_type: "climbing",
        price: 35,
        start_date: "10-19-2018",
        end_date: "10-21-2018",
        description:
          "This October 19-21, head on down to the best limestone sport cragging Colorado has to offer, and with the American Alpine Club and CAMP USA, and be treated to a weekend of climbing, community, clinics, contests, beer, slideshows, films, giveaways, stewardship and fun.",
        event_url: "https://americanalpineclub.org/shelf-road-cc",
        image_url:
          "https://static1.squarespace.com/static/55830fd9e4b0ec758c892f81/57bc7284f7e0abe0efbb523e/59417dc21b631bad37b19a74/1497464437416/_MG_1604.jpg",
        video_url: "https://www.youtube.com/embed/Cnjv_Brs2Bo"
      }
    },
    {
      id: "6",
      type: "event",
      attributes: {
        name: "Leavenworth Rock Fest",
        city: "Canon Springs",
        state: "CO",
        event_type: "climbing",
        price: 35,
        start_date: "10-19-2018",
        end_date: "10-21-2018",
        description:
          "This October 19-21, head on down to the best limestone sport cragging Colorado has to offer, and with the American Alpine Club and CAMP USA, and be treated to a weekend of climbing, community, clinics, contests, beer, slideshows, films, giveaways, stewardship and fun.",
        event_url: "https://americanalpineclub.org/shelf-road-cc",
        image_url:
          "https://static1.squarespace.com/static/55830fd9e4b0ec758c892f81/57bc7284f7e0abe0efbb523e/59417dc21b631bad37b19a74/1497464437416/_MG_1604.jpg",
        video_url: "https://www.youtube.com/embed/Cnjv_Brs2Bo"
      }
    },
    {
      id: "3",
      type: "event",
      attributes: {
        name: "Joes Valley Fest",
        city: "Canon Springs",
        state: "CO",
        event_type: "climbing",
        price: 35,
        start_date: "10-19-2018",
        end_date: "10-21-2018",
        description:
          "This October 19-21, head on down to the best limestone sport cragging Colorado has to offer, and with the American Alpine Club and CAMP USA, and be treated to a weekend of climbing, community, clinics, contests, beer, slideshows, films, giveaways, stewardship and fun.",
        event_url: "https://americanalpineclub.org/shelf-road-cc",
        image_url:
          "https://static1.squarespace.com/static/55830fd9e4b0ec758c892f81/57bc7284f7e0abe0efbb523e/59417dc21b631bad37b19a74/1497464437416/_MG_1604.jpg",
        video_url: "https://www.youtube.com/embed/Cnjv_Brs2Bo"
      }
    }
  ]
};

export const mockUser = {
  name: 'Jane Doe',
  image_url: 'https://alumni.turing.io/sites/default/files/styles/graduate_full_profile/public/squareheadshot1.jpg?itok=EGlR-8YR',
  bio: 'Hi I’m Jane and I have no friends because this app only has one user, and it’s me!!! Hi I’m Jane and I have no friends because this app only has one user, and it’s me!!! Hi I’m Jane and I have no friends because this app only has one user, and it’s me!!! Hi I’m Jane and I have no friends because this app only has one user, and it’s me!!!'
}

export const mockUserEvents = [
  {
    id: "6",
    type: "event",
    attributes: {
      name: "Leavenworth Rock Fest",
      city: "Canon Springs",
      state: "CO",
      event_type: "climbing",
      price: 35,
      start_date: "10-19-2018",
      end_date: "10-21-2018",
      description:
        "This October 19-21, head on down to the best limestone sport cragging Colorado has to offer, and with the American Alpine Club and CAMP USA, and be treated to a weekend of climbing, community, clinics, contests, beer, slideshows, films, giveaways, stewardship and fun.",
      event_url: "https://americanalpineclub.org/shelf-road-cc",
      image_url:
        "https://static1.squarespace.com/static/55830fd9e4b0ec758c892f81/57bc7284f7e0abe0efbb523e/59417dc21b631bad37b19a74/1497464437416/_MG_1604.jpg",
      video_url: "https://www.youtube.com/embed/Cnjv_Brs2Bo",
      status: 'attending'
    }
  },
  {
    id: "3",
    type: "event",
    attributes: {
      name: "Joes Valley Fest",
      city: "Canon Springs",
      state: "CO",
      event_type: "climbing",
      price: 35,
      start_date: "10-19-2018",
      end_date: "10-21-2018",
      description:
        "This October 19-21, head on down to the best limestone sport cragging Colorado has to offer, and with the American Alpine Club and CAMP USA, and be treated to a weekend of climbing, community, clinics, contests, beer, slideshows, films, giveaways, stewardship and fun.",
      event_url: "https://americanalpineclub.org/shelf-road-cc",
      image_url:
        "https://static1.squarespace.com/static/55830fd9e4b0ec758c892f81/57bc7284f7e0abe0efbb523e/59417dc21b631bad37b19a74/1497464437416/_MG_1604.jpg",
      video_url: "https://www.youtube.com/embed/Cnjv_Brs2Bo",
      status: 'wishlist'
    }
  }
]