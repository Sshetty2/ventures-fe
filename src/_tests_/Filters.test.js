import React from 'react';
import { shallow } from 'enzyme';
import { Filters, mapStateToProps } from '../containers/Filters/Filters';
import { mockEvents } from '../data/mockData';

const mockProps = {
  events: mockEvents.data
};

describe('Filters', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Filters {...mockProps} />)
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('mapStateToProps', () => {
    it('should return a props object with events', () => {
      const mockState = {
        events: mockEvents.data,
        isLoading: true
      };
      const expectedState = {
        events: mockEvents.data
      };
      const mappedProps = mapStateToProps(mockState);
      expect(mappedProps).toEqual(expectedState);
    });
  });
});