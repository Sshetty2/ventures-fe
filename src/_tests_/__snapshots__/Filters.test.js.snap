// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filters should match the snapshot 1`] = `
<div
  className="Filters"
>
  <h3
    className="filters-title"
  >
    FILTER EVENTS
  </h3>
  <select
    name="event_type"
    onChange={[Function]}
    value=""
  >
    <option
      key="Type"
      value=""
    >
      Event Type
    </option>
    <option
      key="climbing"
      value="climbing"
    >
      Climbing
    </option>
  </select>
  <select
    name="month"
    onChange={[Function]}
    value=""
  >
    <option
      key="Month"
      value=""
    >
      Month
    </option>
    <option
      key="January"
      value="January"
    >
      January
       (
      0
      )
    </option>
    <option
      key="February"
      value="February"
    >
      February
       (
      0
      )
    </option>
    <option
      key="March"
      value="March"
    >
      March
       (
      0
      )
    </option>
    <option
      key="April"
      value="April"
    >
      April
       (
      0
      )
    </option>
    <option
      key="May"
      value="May"
    >
      May
       (
      0
      )
    </option>
    <option
      key="June"
      value="June"
    >
      June
       (
      0
      )
    </option>
    <option
      key="July"
      value="July"
    >
      July
       (
      0
      )
    </option>
    <option
      key="August"
      value="August"
    >
      August
       (
      0
      )
    </option>
    <option
      key="September"
      value="September"
    >
      September
       (
      0
      )
    </option>
    <option
      key="October"
      value="October"
    >
      October
       (
      7
      )
    </option>
    <option
      key="November"
      value="November"
    >
      November
       (
      0
      )
    </option>
    <option
      key="December"
      value="December"
    >
      December
       (
      0
      )
    </option>
  </select>
  <select
    name="state"
    onChange={[Function]}
    value=""
  >
    <option
      key="State"
      value=""
    >
      State
    </option>
    <option
      key="CO"
      value="CO"
    >
      CO
    </option>
  </select>
</div>
`;
